### PR TITLE
SONAR-6382 add Java ws to update a plugin

### PR DESCRIPTION
**Pull request by sns-seb**
updates a plugin to its latest compatible version

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UpdatePluginsWsAction.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  public UpdatePluginsWsAction(UpdateCenterMatrixFactory updateCenterFactory, PluginDownloader pluginDownloader) {
+    this.updateCenterFactory = updateCenterFactory;
+    this.pluginDownloader = pluginDownloader;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    WebService.NewAction action = controller.createAction("update")
+      .setPost(true)
+      .setDescription("Updates a plugin specified by its key to the latest version compatible with the SonarQube instance")
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UpdatePluginsWsAction.java]: I think update center should be mentioned.

File: server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UpdatePluginsWsAction.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      .setHandler(this);
+
+    action.createParam(PARAM_KEY)
+      .setRequired(true)
+      .setDescription("The key identifying the plugin to update");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UpdatePluginsWsAction.java]: Security - check that user is a system administrator.

File: server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UpdatePluginsWsAction.java (hunk 1 above)
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UpdatePluginsWsAction.java]: how so ?

Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UpdatePluginsWsAction.java]: I don't know and you ? :-)

**Comment by sns-seb** [comment]
accidentally merged into master, blocking comment taken into account in #251

File: server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UpdatePluginsWsAction.java (hunk 1 above)
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UpdatePluginsWsAction.java]: I'll update description of all WS using the update center when I'll add the updatecenter timestamp

**Pull request merged**